### PR TITLE
Expose Metadata

### LINKED ISSUE #133: Expose metadata endpoints
api/guilds/:id has metadata for servers (this seems to require you're joined to the server) (this also has full role info)
api/guilds/:id/members/:id can be used to get member metadata, this might be useful for non-ws usage of the lib. (this requires you are joined to the server)
api/guilds/:id/channels - useful for non-ws (requires you're joined to server)

/api/v6/users/@me/guilds - some others too

### PR DESCRIPTION
**Pull request by NCPlayz**
### Summary

Added a few metadata endpoints, which resolves #133. 

### Usage

- `Client.fetch_guild`: ``/guilds/{guild_id}``

```py
>>> g = await bot.fetch_guild(81384788765712384)
... await ctx.send(g.name)
Discord API
```

- `Guild.fetch_member`: ``/guilds/{guild_id}/members/{member_id}``

```py
>>> m = await g.fetch_member(80088516616269824)
... await ctx.send(m.created_at)
2015-08-10 00:03:10.334000
```

- `Client.fetch_guilds`: ``/users/@me/guilds``

```py
>>> async for guild in bot.fetch_guilds(limit=10, after=discord.Object(id=121678432504512512)):
...     await ctx.send(guild.name)
aaaaaaaaaaaaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaa
```

### BREAKING CHANGE

The only breaking change included with this pull request is the renaming of `Client.get_user_info` to `Client.fetch_user`. This was done to match the naming scheme as suggested [here](https://github.com/Rapptz/discord.py/issues/133#issuecomment-471733377).

### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

I wasn't sure if the documentation was sufficient for this, so it would be very useful if it could be reviewed.

### CODE REVIEW

**Comment by gamescom15** [comment]
If `get_user_info` is renamed to `fetch_user_info`, renaming `get_user_profile` should also be considered.

The problem with this naming scheme is that there are a few more functions named `get_...` that have to be awaited, for example `get_message` and `get_ban` and many more.

**Comment by NCPlayz** [comment]
I did consider this and discussed how it should be done, and it was deemed extensive. I concluded that `fetch` should only be used for those four endpoints because they respond back with metadata.

**Comment by khazhyk** [comment]
Drop the `_info`, which was previously only used for disambiguation?

([Danny agrees](https://canary.discordapp.com/channels/336642139381301249/336642776609456130/554847093785886723))

**Comment by NCPlayz** [comment]
Update:
- I have removed the `_info` suffix for `fetch_guild`, `fetch_member`, and `fetch_user`.
- I have moved `fetch_member` to `Guild` to save an API call.
- I have renamed `HTTPClient.get_all_guilds` to `HTTPClient.get_guilds`.
- I added a note to `fetch_guild`'s limit parameter (`Defaults to 100.`).
- Removed a no-longer-needed note in `Guild.fetch_member`.

File: discord/client.py [hunk 1]
```diff
@@ -954,7 +954,7 @@ def fetch_guilds(self, *, limit=100, before=None, after=None):
 
     # Invite management
 
-    async def get_invite(self, url, *, with_counts=True):
+    async def fetch_invite(self, url, *, with_counts=True):
```
Comment (apple502j) [inline discord/client.py]: This is used in InviteConverter of command extension. Also change it.

File: docs/migrating.rst [hunk 2]
```diff
@@ -148,10 +148,14 @@ A list of these changes is enumerated below.
...
 +---------------------------------------+------------------------------------------------------------------------------+
-| ``Client.get_message``                | :meth:`abc.Messageable.get_message`                                          |
+| ``Client.get_invite``                 | :meth:`Client.fetch_invite`                                                  |
++---------------------------------------+------------------------------------------------------------------------------+
+| ``Client.get_message``                | :meth:`abc.Messageable.fetch_message`                                        |
 +---------------------------------------+------------------------------------------------------------------------------+
 | ``Client.get_reaction_users``         | :meth:`Reaction.users`                                                       |
 +---------------------------------------+------------------------------------------------------------------------------+
+| ``Client.get_user_info``              | :meth:`Client.fetch_user_info`                                               |
```
Comment (apple502j) [inline docs/migrating.rst]: Wasn't it `fetch_user`?

```suggestion
| ``Client.get_user_info``              | :meth:`Client.fetch_user`                                               |
```

**Pull request closed without merging**